### ISSUE TITLE
HomeAway - Occupancy sensor AND switch.

### ISSUE THREAD

**Issue by hjdhjd** [chrisjshull/homebridge-nest#111]
I saw in a recent git rev you’ve got HomeAway configurable as a switch OR as a sensor.

I’d suggest that this shouldn’t be a limitation. I may use the occupancy sensor element in my automations, but what TRIGGERS the sensor would be tripping the HomeAway switch itself, if that makes sense.

Put another way...automations may/do rely on the occupancy sensor as a conditional trigger, however that trigger is set off by flipping the HomeAway switch.

In my use case, for instance...HomeKit seems to have better geolocation capabilities than the Nest app, and with pets in a household it’s problematic to let Nest “figure out” when one is Home/Away. Creating an automation to flip the switch to Away when the last one leaves the house, and to Home when the first arrives solves that problem.

Additionally, if there are automations that rely on Home/Away status, they can be triggered by the occupancy sensor.

TLDR: suggest you allow people to have BOTH a Home/Away occupancy sensor AND a Home/Away switch, rather than limit it in one direction.

**Comment by adriancable** [comment]
It's definitely possible but I want to make sure I understand your use case scenario before I add complexity via additional options.

Reading through this a few times I'm not 100% sure I understand what you're trying to do, so maybe you can give me some more detail. From what I understand, you're trying to use HomeKit's geolocation to set Home/Away. But you can do this today by creating a pair of automations - the first 'When Anyone Arrives Home' to turn the 'Home/Away' switch to On, and the second 'When Anyone Leaves Home' to turn the 'Home/Away' switch to Off. It seems this will do what you want, without the need for a duplicate home/away sensor.

But presumably this isn't quite what you want?

**Comment by hjdhjd** [comment]
Adrian - the automation you've described is what I already use, and it works well. I don't have a great use case to share offhand that is a functionally unique and distinct one for sensors versus switches.

I understand the balance of complexity versus utility here, and there may be use cases in the future where sensors are a preferred trigger mechanism to switches for certain automation types. Instinctively a sensor is what I use to detect "transient state" (occupancy is detected versus not), and a switch is something I use to take action "set occupancy to on or off".

My personal sense of symmetry is to say "a Home/Away switch is the **action** we use to toggle between Home and Away" and "a Home/Away occupancy sensor is what we use to **trigger** scenes or as **conditionals**". I don't view these as "either/or" things and some people may want both (switch and sensor), either, or neither. For people like me who use HomeKit to set Home/Away status, the switch is essential...but I also prefer to use the sensor in my other automations that depend on occupancy state. For people who use Nest to set Home/Away, the switch is less important than the sensor, I imagine.

And maybe that's tomorrow's problem and we don't need to worry about it today...that's cool too. It's not that big a deal to switch automations to use switch state instead of sensor state currently...it just bugs me conceptually. :)

Last point I'd raise is what level of consistency you want to have with other HomeKit devices in the same category like Ecobee. Ecobee displays occupancy as sensors and has a Home/Away switch as well, I believe.

**Comment by adriancable** [comment]
OK - the latest commit adds a new HomeAway.AsOccupancySensorAndSwitch option to enable this.

**Comment by rbpasker** [comment]
I have HomeAway.AsOccupancySensorAndSwitch but the sensor triggers on, but not off. is this the desired behavior?

**Comment by adriancable** [comment]
(Reopening) No, that doesn't sound right. The sensor should always follow the state of the switch. Are you saying the switch state is correct, but the occupancy sensor state is not?

**Comment by rbpasker** [comment]
so, when I click the Occupied switch from off to on, the sensor goes on and shows the word "triggered".  when I click the switch from on to off, the sensor goes off, but doesn't show triggered 

![image](https://user-images.githubusercontent.com/87866/65063898-31a33700-d94d-11e9-8d42-271aab0889d6.png)

![image](https://user-images.githubusercontent.com/87866/65063912-3d8ef900-d94d-11e9-8197-d626512d8744.png)

**Comment by FixMiller** [comment]
Hi Adrian, I'm confused about the use case too. I got here by researching if one can use the Nest Thermostat inbuilt motion detector as trigger.
For example turn on some light when I pass in front of it at night. 
Is such a thing possible?

**Comment by adriancable** [comment]
@FixMiller - I don't think this is possible. There is a subtle way to use Nest Protect's built-in motion detector and I do expose this functionality. I will look further into whether this is possible with the Thermostat, but I believe not. Sorry!
